File: Medium/451. Sort Characters By Frequency/solution.cs
public class Solution {
    public string FrequencySort(string s) {
        var freq = new Dictionary<char, int>();
        foreach (var ch in s){
            if (freq.ContainsKey(ch)){
                freq[ch]++;
            }
            else{
                freq.Add(ch, 1);
            }
        }

        var sortedFreq = freq.OrderByDescending(kv => kv.Value);
        var result = new StringBuilder();
        foreach(var kvp in sortedFreq){
            for (int i = 0; i < kvp.Value; i++){
                result.Append(kvp.Key);
            }
        }

        return result.ToString();
    }
}

File: Medium/451. Sort Characters By Frequency/solution.cpp
class Solution {
    vector<int> freq;
public:
    string frequencySort(string s) {
        unordered_map<char, int> freq;
        vector<string> bucket(s.size() + 1, "");
        for (char ch : s){
            freq[ch]++;
        }
        for (auto it : freq){
            bucket[it.second].append(it.second, it.first);
        }
        string result = "";
        for (int i = s.size(); i >= 1; i--){
            if (!bucket[i].empty()){
                result += bucket[i];
            }
        }
        return result;
    }
};